packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1034,7 +1039,29 @@ export class SnapController extends BaseController<
       }
 
       this._recordSnapRpcRequest(snapName);
-      return handler(origin, request);
+
+      // handle max request time
+      let resolve: any;
+      let reject: any;
```

Comment:
These can and should be generic functions, something like `(...args: unknown[]) => void`.